dev/fattest.simplicity/src/componenttest/containers/ArtifactoryRegistry.java

```diff
@@ -121,59 +130,56 @@ private static File generateDockerConfig(String registry, String authToken, File
-                int authIndex = contents.indexOf("\"auth\":", registryIndex);
-                int authIndexEnd = contents.indexOf(',', authIndex) + 1;
-                String authSubstring = contents.substring(authIndex, authIndexEnd);
-                if (authSubstring.contains(authToken)) {
-                    Log.info(c, m, "Config already contains the correct auth token for registry: " + registry);
-                    return configFile;
-                } else {
-                    replacedAuth = true;
-                    Log.info(c, m, "Replacing auth token for registry: " + registry);
-                    contents = contents.replace(authSubstring, "\"auth\": \"" + authToken + "\",");
-                }
-            }
+            root = (ObjectNode) mapper.readTree(configFile);
 
-            if (authsIndex >= 0 && !replacedAuth) {
-                Log.info(c, m, "Other auths exist. Need to add auth token for registry: " + registry);
-                int splitAt = contents.indexOf('{', authsIndex);
-                String firstHalf = contents.substring(0, splitAt + 1);
-                String secondHalf = contents.substring(splitAt + 1);
-                contents = firstHalf + '\n' + privateAuth + "," + secondHalf;
-            } else if (!replacedAuth) {
-                Log.info(c, m, "No auths exist. Adding auth block");
-                int splitAt = contents.indexOf('{');
-                String firstHalf = contents.substring(0, splitAt + 1);
-                String secondHalf = contents.substring(splitAt + 1);
-                String delimiter = secondHalf.contains(":") ? "," : "";
-                contents = firstHalf + "\n\t\"auths\": {\n" + privateAuth + "\n\t}" + delimiter + secondHalf;
-            }
+            Log.info(c, m, "Config already exists at: " + configFile.getAbsolutePath());
+            logConfigContents(m, "Original contents", serializeOutput(mapper, root));
         } else {
+            root = mapper.createObjectNode();
+
             configDir.mkdirs();
-            Log.info(c, m, "Generating a private registry config file at: "
-                           + configFile.getAbsolutePath());
-            contents = "{\n\t\"auths\": {\n" + privateAuth + "\n\t}\n}";
+            Log.info(c, m, "Generating a private registry config file at: " + configFile.getAbsolutePath());
         }
-        logConfigContents(m, "New config.json contents are", contents);
-        configFile.delete();
-        writeFile(configFile, contents);
+
+        //If existing config contains correct registry and auth token combination, return original file.
+        try {
+            if (root.get("auths").get(registry).get("auth").textValue().equals(authToken)) {
+                Log.info(c, m, "Config already contains the correct auth token");
+                return configFile;
+            }
+        } catch (Exception ignore) {
+            //assume config did not contain authToken
```

Comment:
Should we still log the exception here even if it's ignored, for potential debugging?